include/RAJA/policy/openmp/target_reduce.hpp

```diff
@@ -117,8 +117,8 @@ struct Reduce_Data {
   explicit Reduce_Data(T defaultValue, T identityValue, Offload_Info &info)
       : value(identityValue),
         device{reinterpret_cast<T *>(
-            omp_target_alloc(Teams * sizeof(T), info.deviceID))},
-        host{new T[Teams]}
+            omp_target_alloc(Threads * sizeof(T), info.deviceID))},
+        host{new T[Threads]}
```

Comment:
I think this will break the omp target reducers when `Teams > Threads`
Each team writes one value to this array, so when there there are more teams than threads per team this will cause writes beyond the end of the buffer.

Reply:
@MrBurmark makes a great point. However, the code in its current form will guarantee that Teams <= Threads, but it is not obvious to RAJA developers (and less obvious to RAJA users). The guarantee is specified in the omp_target_parallel_for_exec pattern (/include/RAJA/policy/openmp/target_forall.hpp):

`auto teamnum = RAJA_DIVIDE_CEILING_INT( (int)distance, (int)Threads );`
`#pragma omp target teams distribute parallel for num_teams(teamnum) thread_limit(Threads) schedule(static, 1) map(to : body)`

So far, this is the only pattern we use for omp_target policies, but if we develop other policies they will require similar team computations or checks. There is another unused pattern (omp_target_parallel_for_exec_nt) in which neither teams nor threads are specified. But this should still work with reduce because the default number of teams is 1.

Potentially adding to the confusion is a piece of code I neglected to change:

https://github.com/LLNL/RAJA/blob/2fddee34a2675b999bd141835ff0efe12e7ac08a/include/RAJA/policy/openmp/policy.hpp#L182-L187

"Teams" should be changed to "Threads", because that is how they are used with my other modifications.

Reply:
Won't teamnum be greater than Threads if `distance > Threads*Threads`?

Reply:
That is true! I suppose I can add an assert after that calculation, and maybe to be ultra safe, another assert in ~TargetReduce() and ~TargetReduceLoc() where teams are used during reduction.